lenient/src/demo.c

```diff
@@ -0,0 +1,84 @@
+#include "demo.h"
+#include "module.h"
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+static int add_to_foo(int operand);
+
+// Function context begin
+static struct Demo* this_public;
+static struct DemoPrivate* this_private;
+// Function context end
+
+MODULE_GENERATE_CONTEXT_FUNCTIONS(struct Demo, struct DemoFull)
+
+struct Demo*
```

Comment:
Please explain details of function in a comment. The function allocates DemoFull, but returns Demo.

Reply:
Added explanation in: daac5412c00b8e6b0a44a3cca7bd2e949e05433d